packages/aila/src/core/chat/AilaStreamHandler.ts

```diff
@@ -89,9 +90,20 @@ export class AilaStreamHandler {
       type: "comment",
       value: "CHAT_START",
     });
-    const messages = this._chat.completionMessages();
-    this._streamReader =
-      await this._chat.createChatCompletionObjectStream(messages);
+    const currentLessonPlan = this._chat.aila.lessonPlan;
+
+    const AILA_ENGINE_STREAM_URL = process.env.AILA_ENGINE_STREAM_URL;
+
+    const lastUserMessage = this._chat.messages.slice(-1)[0];
+    if (AILA_ENGINE_STREAM_URL) {
+      this._streamReader = await createStreamReaderFromURL(
+        `${AILA_ENGINE_STREAM_URL}?prompt=${JSON.stringify(currentLessonPlan)}\n\n${lastUserMessage?.content}`,
```

Comment:
Currently, this is super naive. Just sends the current lesson plan concatenated with the last user message to the server. i.e. not the full message history.

Plan to amend that on a subsequent update.